x/ecocredit/server/testsuite/suite.go

```diff
@@ -635,4 +635,76 @@ func (s *IntegrationTestSuite) TestScenario() {
 			}
 		})
 	}
+
+	/****   TEST ALLOWLIST CREDIT DESIGNERS   ****/
```

Comment:
We've not really got many precedents for this, but this is quite a good spot for some property-based testing using something like [gopter](https://pkg.go.dev/github.com/leanovate/gopter). You should be able to codify the invariants from below and get more certainty that we're doing things right. I'll approve though, as this is really a nice to have.

Reply:
neat - haven't heard of this before. i will look into it - thanks for the suggestion 👍🏻

Reply:
Cool, I've used it quite a lot in Haskell and done a training course even, so let me know if you need any advice!

Reply:
Have you compared [rapid](https://github.com/flyingmutant/rapid) and gopter @ruhatch ? rapid bills itself as simpler

Reply:
> Have you compared [rapid](https://github.com/flyingmutant/rapid) and gopter @ruhatch ? rapid bills itself as simpler

Ah, not yet, I'll give them a quick look over

Reply:
@technicallyty @aaronc yeah, `rapid` does look quite nice, so maybe give that a go!

Reply:
It would be good if we can align on a package to use in both regen-ledger and the SDK. I've used gopter in the past and it worked but it was a bit complex. Does rapid look like the clear winner @ruhatch or should we try and see what we think?

Reply:
Also, if we start adopting property based testing it would be good to do a webinar style tutorial for the whole dev team because I think it's a paradigm that takes a bit of getting used to

Reply:
yeah that would be helpful, i looked at it for a bit and wasnt really sure where to start 🤷🏻

Reply:
@aaronc @technicallyty yeah, some sort of webinar sounds like a good idea!

I could try and do some property testing on this one, trying with both `rapid` and `gopter` to see which one comes out nicer and feels easier to use, then we can make a better decision. How does that sound? If that's good, perhaps we can make a new issue for property testing the credit class permissions, assigned to me, and merge this PR as is?

Reply:
+1 for that @ruhatch - would love to attend something to learn more